src/data/simple_dmatrix.cc

```diff
@@ -147,8 +147,10 @@ SimpleDMatrix::SimpleDMatrix(AdapterT* adapter, float missing, int nthread) {
     info_.num_col_ = adapter->NumColumns();
   }
 
-  // Synchronise worker columns
-  rabit::Allreduce<rabit::op::Max>(&info_.num_col_, 1);
+  // Check each distributed partition has the same number of features
+  uint64_t max_column = info_.num_col_;
+  rabit::Allreduce<rabit::op::Max>(&max_column, 1);
+  CHECK_EQ(max_column, info_.num_col_);
```

Comment:
The `num_col_` could be 0.

Reply:
I don't understand. This 1 refers to the size of the rabit send buffer, as opposed to an operand of max.

Reply:
@RAMitchell When input data is empty, `info.num_col_` equals to 0.  `CHECK_EQ(max_column, 0)` will fail.

Reply:
Ah I see, in the empty partition case. In any case I remove this code and tried something different.